owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,365 @@
+
+import com.owncloud.android.data.OwncloudDatabase
+import com.owncloud.android.data.transfers.datasources.implementation.OCLocalTransferDataSource
+import com.owncloud.android.data.transfers.datasources.implementation.OCLocalTransferDataSource.Companion.toEntity
+import com.owncloud.android.data.transfers.datasources.implementation.OCLocalTransferDataSource.Companion.toModel
+import com.owncloud.android.data.transfers.db.OCTransferEntity
+import com.owncloud.android.data.transfers.db.TransferDao
+import com.owncloud.android.domain.camerauploads.model.UploadBehavior
+import com.owncloud.android.domain.transfers.model.OCTransfer
+import com.owncloud.android.domain.transfers.model.TransferResult
+import com.owncloud.android.domain.transfers.model.TransferStatus
+import com.owncloud.android.domain.transfers.model.UploadEnqueuedBy
+import com.owncloud.android.testutil.OC_ACCOUNT_NAME
+import io.mockk.every
+import io.mockk.mockkClass
+import io.mockk.verify
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class OCLocalTransferDataSourceTest {
+    private lateinit var ocLocalTransferDataSource: OCLocalTransferDataSource
+    private val transferDao = mockkClass(TransferDao::class)
+    private val id = -1L
+    private val ocTransfer = OCTransfer(
+        id = id,
+        localPath = "/local/path",
+        remotePath = "/remote/path",
+        accountName = OC_ACCOUNT_NAME,
+        fileSize = 1024L,
+        status = TransferStatus.TRANSFER_IN_PROGRESS,
+        localBehaviour = UploadBehavior.MOVE,
+        forceOverwrite = true,
+        createdBy = UploadEnqueuedBy.ENQUEUED_BY_USER
+    )
+
+    private var transferEntity: OCTransferEntity = ocTransfer.toEntity()
+
+    @Before
+    fun setUp() {
+        val db = mockkClass(OwncloudDatabase::class)
+
+        every { db.transferDao() } returns transferDao
+
+        ocLocalTransferDataSource = OCLocalTransferDataSource(transferDao)
+    }
+
+    @Test
+    fun `saveTransfer returns a Long`() {
+        val resultExpected = 1L
+        every {
+            transferDao.insertOrReplace(any())
+        } returns resultExpected
+
+        val resultActual = ocLocalTransferDataSource.saveTransfer(ocTransfer)
+
+        assertEquals(resultExpected, resultActual)
+
+        verify(exactly = 1) {
+            transferDao.insertOrReplace(ocTransfer.toEntity())
+        }
+    }
+
+    @Test
+    fun `updateTransfer returns a Long`() {
+        val resultExpected = 1L
+        every {
+            transferDao.insertOrReplace(any())
+        } returns resultExpected
+
+        ocLocalTransferDataSource.updateTransfer(ocTransfer)
+
+        verify(exactly = 1) {
+            transferDao.insertOrReplace(ocTransfer.toEntity())
+        }
+    }
+
+    @Test
+    fun updateTransferStatusToInProgressById() {
```

Comment:
Name them correctly with the convention we follow

Reply:
Done